### PR TITLE
Add cmakelists

### PR DESCRIPTION
**Pull request by rajkan01**
- Added CMake and Travis CI support
- Updated `mbed-os.lib` to pick the latest master
- Updated `wifi-ism43362.lib` to latest to pick CMake changes

@0xc0170 @hugueskamba @linlingao @evedon

### CODE REVIEW

File: mbed-os.lib [hunk 1]
```diff
@@ -1 +1 @@
-https://github.com/ARMmbed/mbed-os/#0db72d0cf26539016efbe38f80d6f2cb7a3d4414
+https://github.com/ARMmbed/mbed-os/
```
Comment (LDong-Arm) [inline mbed-os.lib:1]: Un-versioned mbed-os.lib is not compatible with the current release script yet, if I remember correctly.

For now, just rebase this PR to the latest development branch so `mbed-os.lib` points to the latest 6.5.0 release which is new enough.

**Review by LDong-Arm — CHANGES_REQUESTED** [review]
Thanks @rajkan01, I've merged https://github.com/ARMmbed/mbed-client-for-aws/pull/18 so feel free to update the client's .lib file.

File: mbed-os.lib (hunk 1 above)
Comment (LDong-Arm) [reply mbed-os.lib:1]: Okay, didn't know the script is updated.

**Review by LDong-Arm — APPROVED** [review]
LGTM. Up to the CMake team to review the CI script.

**Comment by LDong-Arm** [comment]
Please rebase to the latest development, to dismiss the merge conflict on GitHub

**Review by 0xc0170 — APPROVED** [review]

**Pull request merged**